src/inmanta/ast/statements/call.py

```diff
@@ -261,13 +262,14 @@ def __init__(self, queue_scheduler, resolver, result: ResultVariable, requires,
         self.ready(self)
 
     def execute(self):
-        requires = {k: v.get_value() for (k, v) in self.requires.items()}
+        requires = {k: v.get_value() for (k, v) in self.base_requires.items()}
         try:
             self.function.resume(requires, self.resolver, self.queue_scheduler, self.result)
             self.done = True
         except UnsetException as e:
             LOGGER.debug("Unset value in python code in plugin %s %s.%s.", self.function.function, e.instance, e.attribute)
-            self.waitfor(e.get_result_variable())
+            # Don't handle it here!
```

Comment:
Maybe add an explanation why this shouldn't be handled here.

Reply:
seem to have forgotten to push my changes before